packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -142,30 +164,44 @@ function makeVisible(): void {
 
 function processInput(): void {
   loading = true;
-  searchFunction(value)
+  items = [];
+  itemHeadings = {};
+  onInputChange(value)
     .then(result => {
       // if the component has been disabled in the meantime
       if (disabled) {
         return;
       }
-      items = result.toSorted((a: string, b: string) => {
-        const dockerIoValue = `docker.io/${value}`;
-        const aStartsWithValue = a.startsWith(value) || a.startsWith(dockerIoValue);
-        const bStartsWithValue = b.startsWith(value) || b.startsWith(dockerIoValue);
-        if ((aStartsWithValue && bStartsWithValue) || (!aStartsWithValue && !bStartsWithValue)) {
-          return a.localeCompare(b);
-        } else if (aStartsWithValue && !bStartsWithValue) {
-          return -1;
-        } else {
-          return 1;
+      for (let { values, group, sorted } of result) {
```

Comment:
Following https://github.com/podman-desktop/podman-desktop/pull/10573/files#r1912902298 you should be able to use a derived state for the `groupedValues` (or better naming to be found) instead of having logic in the processInput